src/chapters/unit-testing/hello-jasmine.rst

```diff
@@ -0,0 +1,301 @@
+Hello, Jasmine!
+===============
+
+.. index:: ! Jasmine
+
+In order to unit test our code, we need to use a module. Such a module of called a **unit-testing framework**, **test runner**, or **test harnesss**, and there are `many to choose from <https://en.wikipedia.org/wiki/List_of_unit_testing_frameworks#JavaScript>`_. 
+
+We will use `Jasmine <https://jasmine.github.io/>`_, a popular JavaScript testing framework. 
+
+Using Jasmine
+-------------
+
+.. index:: 
+   single: Jasmine; installing
+
+Installing Jasmine with NPM is simple:
+
+::
+
+   $ npm install -g jasmine
+   /usr/local/bin/jasmine -> /usr/local/lib/node_modules/jasmine/bin/jasmine.js
+   + jasmine@3.4.0
+   updated 3 packages in 0.425s
+
+The ``-g`` flag install Jasmine *globally*, as opposed to *locally* (for only the current project directory).
+
+Using repl.it removes the need to install Jasmine, since it will do so automatically. 
+
+.. admonition:: Try It!
+
+   Run some `tests for the reverse function <https://repl.it/@launchcode/reverse-Function-With-Tests>`_. This is the same ``reverse`` function that :ref:`we wrote previously <reverse_func>`. 
+   
+   Don't worry about understanding the code at this point, just hit *run* to execute the tests. How many total tests are there? How many passed? How many failed? 
+
+.. index::
+   single: Jasmine; project structure
+
+A project using Jasmine has several components. Here's the project structure:
+
+.. figure:: figures/jasmine-project-structure.png
+   :alt: A small project with three important files: index.js, reverse.js, reverse.spec.js
+
+   A Jasmine project
+
+There are three important files:
+
+#. ``reverse.js`` contains the ``reverse`` function, which must be exported for use in other files.
+#. ``spec/reverse.spec.js`` contains the tests for ``reverse``.
+#. ``index.js`` contains the Jasmine code needed to run the tests. This is the file that executes when you hit *run* in repl.it.
+
+.. warning:: Jasmine can be set up and used in many different ways. If you are looking for an answer on the Internet,---on Stack Overflow or in the Jasmine documentation---you will see widely varying usages of Jasmine that don't apply to your situation. Rely on this book as your main reference, and you'll be fine.
+
+Hello, Jasmine!
+---------------
+
+Let's build a "Hello, World!" Jasmine project, to get familiar with the basic components. Open and fork `this repl.it project <https://repl.it/@launchcode/Hello-Jasmine-Starter-Code>`_
+
+We will walk you through the steps need to get a simple Jasmine project up and running. Code along with us throughout this section.
+
+
+``index.js``
+^^^^^^^^^^^^
+
+This is the main project file. Up until now, ``index.js`` is where you have been writing the code for a given exercise or assignment. Now that we are writing tests for our code, ``index.js`` will contain the Jasmine code to find and execute the tests. Our project-specific code will live in other files.
+
+.. sourcecode:: js
```

Comment:
The repl.it code does not match the sourcecode below.  In particular, repl.it shows ``var`` instead of ``const``.